File: Week_01/G20200343040009/homework/src/com/studyWeek2/LeetCode242.java
package com.studyWeek2;

/**
 * 1. 长度不等，false
 * 2. 若相等，初始化26个字母表的hash表，遍历s、t字符串
 * 3. 利用一个计数器做加减
 */
public class LeetCode242 {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        int[] counter = new int[26];
        for (int i = 0; i < s.length(); i++) {
            counter[s.charAt(i) - 'a']++;
            counter[t.charAt(i) - 'a']--;
        }
        for (int count : counter) {
            if (count != 0) {
                return false;
            }
        }
        return true;
    }
}

File: Week_08/G20200343040039/LeetCode_242.go
package G20200343040039

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	counter := make([]byte, 26, 26)
	for i := 0; i < len(s); i++ {
		counter[s[i] - 'a']++
		counter[t[i] - 'a']--
	}

	for i := 0; i < len(counter); i++ {
		if counter[i] != 0 {
			return false
		}
	}

	return true
}